hotness_schema/hotness_schema/messages.py

```diff
@@ -72,39 +51,68 @@ def summary(self):
         """
         if self.reason == "anitya":
             return (
-                "the-new-hotness saw an update for '{}', ".format(self.project)
+                "the-new-hotness saw an update for '{}', ".format(self.package)
                 + "but release-monitoring.org doesn't know what that project is called "
                 + "in Fedora land"
             )
         elif self.reason == "rawhide":
             return (
-                "the-new-hotness saw an update for '{}', ".format(self.project)
+                "the-new-hotness saw an update for '{}', ".format(self.package)
                 + "but no rawhide version of the package could be found yet"
             )
         elif self.reason == "pkgdb":
             return (
-                "the-new-hotness saw an update for '{}', ".format(self.project)
+                "the-new-hotness saw an update for '{}', ".format(self.package)
                 + "but pkgdb says the maintainers are not interested in bugs being filed"
             )
         elif self.reason == "bugzilla":
             return (
-                "the-new-hotness saw an update for '{}', ".format(self.project)
+                "the-new-hotness saw an update for '{}', ".format(self.package)
                 + "but the bugzilla issue couldn't be updated"
             )
         else:
             return "the-new-hotness saw an update for '{}', ".format(
-                self.project
+                self.package
             ) + "but it got dropped for reason: '{}'".format(self.reason)
 
     @property
-    def project(self):
+    def package(self):
         """
         Return a name of the project.
 
         Returns:
              (str): Name of the project.
         """
-        return self.body["trigger"]["msg"]["project"]
+        if self.reason == "anitya":
+            original = self.body["trigger"]["msg"]
+
+            packages = None
+            if "packages" in original["message"]:
+                packages = original["message"]["packages"]
+            elif "packages" in original:
+                packages = original["packages"]
+
+            if packages:
+                return set(
+                    [
+                        pkg["package_name"]
+                        for pkg in packages
+                        if pkg["distro"] == "Fedora"
+                    ]
+                ).pop()
+
+        if "package_listing" in self.body["trigger"]["msg"]:
+            original = self.body["trigger"]["msg"]
+            return original["package_listing"]["package"]["name"]
+
+        if "buildsys.build" in self.body["trigger"]["topic"]:
+            return self.body["trigger"]["msg"]["name"]
+
+        if "package" in self.body["trigger"]["msg"]:
+            original = self.body["trigger"]["msg"]
+            return original["package"]["name"]
+
+        return ""
```

Comment:
Are you sure you want to return an empty string if the package can't be found? Returning `None` seems more appropriate, but it's up to you.

Also, it'd be nice to redefine the `packages()` method to return a list containing only the result of the `package()` method (if not empty / not `None`)

Reply:
This is the same logic, that the fedmsg_meta_fedora_infrastructure is using.

It will be probably better to override `packages()` property from `Message`.